infra/events/src/main/java/com/simprints/infra/events/event/domain/models/OneToOneMatchEvent.kt

```diff
@@ -22,40 +24,82 @@ data class OneToOneMatchEvent(
         matcher: String,
         result: MatchEntry?,
         fingerComparisonStrategy: FingerComparisonStrategy?,
+        probeBiometricReferenceId: String,
     ) : this(
-        UUID.randomUUID().toString(),
-        OneToOneMatchPayload(
+        id = UUID.randomUUID().toString(),
+        payload = OneToOneMatchPayload.OneToOneMatchPayloadV4(
             createdAt = createdAt,
             eventVersion = EVENT_VERSION,
             endedAt = endTime,
             candidateId = candidateId,
             matcher = matcher,
             result = result,
             fingerComparisonStrategy = fingerComparisonStrategy,
+            probeBiometricReferenceId = probeBiometricReferenceId,
         ),
-        ONE_TO_ONE_MATCH,
+        type = ONE_TO_ONE_MATCH,
     )
 
     override fun getTokenizableFields(): Map<TokenKeyType, TokenizableString> = emptyMap()
 
     override fun setTokenizedFields(map: Map<TokenKeyType, TokenizableString>) = this // No tokenized fields
 
+    @JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.EXISTING_PROPERTY,
+        property = "eventVersion",
+        visible = true,
+    )
+    @JsonSubTypes(
+        JsonSubTypes.Type(
+            value = OneToOneMatchPayload.OneToOneMatchPayloadV3::class,
+            name = VERSION_WITHOUT_REFERENCE_ID.toString(),
```

Comment:
Probably a stupid question but in why do we need the old version?

Reply:
We are storing those models as String in the event database, so the old version should still be un-marshallable even after SID upgrades. And migrating data in those JSON strings isn't really possible since we don't have the correct reference ID for each (at least within a reasonable effort).

Reply:
Aren't we uploading the JSONs directly as strings? 
Or is the un-marshalling needed for the new diagnostic screen that shows events?

Reply:
Not as string, there is a mapping from domain models to API models in between.

Reply:
To be precise: from DB model string field to domain model to API model. :D